### DISCUSSION TITLE
[UX] reapply offset of duplicated and relocated objects to following duplications

### DISCUSSION THREAD

**Discussion by ben-wes** [pure-data/pure-data#2369] [Ideas]
title sounds a bit complicated. here's the scenario where this would apply:
1. select object
2. duplicate via <kbd>ctrl</kbd>+<kbd>d</kbd>
3. move to a new location with arrow keys / mouse
4. duplicate again

currently, the next duplicated object will be placed at offset (10, 10) again. with this feature, it would be placed with the same offset as the previous instance. this offset would only be "remembered" while no other object gets selected.

this is typical behaviour for layout tools like illustrator and figma and might be a nice addition to puredata, too.

**Comment by porres** [comment]
BTW, PlugData is doing that right now (not yet released though)

I agree it is a nice feature!

**Comment by umlaeute** [comment]
not sure i understand.

`<<duplicate>>` duplicate the currently selected object(s). (offsets are relative to the selection).
do I understand correctly, that you would like to break this, and instead have `<<duplicate>>` stick to the first duplicated selection? (which can be changed by explicitly selecting something else).

because in order to *move* the duplication around, it needs to be selected.

**Reply by ben-wes** [reply]
the description above is not very precise concerning the "selection". i wouldn't suggest any change to selection behaviour. the "memorized" change would range from duplication to the next duplication ... illustrating with pictures - what currently happens when duplicating, moving duplicate and duplicating again:

1. ![Screenshot 2024-07-01 at 23 32 41](https://github.com/pure-data/pure-data/assets/1923787/f2e7a94e-1e38-4cac-ab1f-24e8228d6354)
2. ![Screenshot 2024-07-01 at 23 32 45](https://github.com/pure-data/pure-data/assets/1923787/ae089bcc-45b7-4e08-b14b-48291d47f16c) 
3. ![Screenshot 2024-07-01 at 23 32 49](https://github.com/pure-data/pure-data/assets/1923787/3d9df629-f09b-4b72-9f61-62d8b99193eb) <-- manually placed with mouse or keyboard
4. ![Screenshot 2024-07-01 at 23 32 51](https://github.com/pure-data/pure-data/assets/1923787/ec3ce9b2-4fa0-4881-98a8-ec9e7cc3a9f8) <-- result after another <kbd>ctrl</kbd>+<kbd>d</kbd>

with the suggested change:

4. ![Screenshot 2024-07-01 at 23 32 59](https://github.com/pure-data/pure-data/assets/1923787/fbdccf2a-4c49-423f-99c4-ccc36652d3c5) <-- previous offset is applied again

**Reply by umlaeute** [reply]
ah i see, thanks for clarifying.
